Mathlib/GroupTheory/Submonoid/Operations.lean

```diff
@@ -469,259 +520,304 @@ namespace OneMemClass
 
 variable {A M₁ : Type _} [SetLike A M₁] [One M₁] [hA : OneMemClass A M₁] (S' : A)
 
-include hA
-
 /-- A submonoid of a monoid inherits a 1. -/
-@[to_additive "An `add_submonoid` of an `add_monoid` inherits a zero."]
-instance hasOne : One S' :=
+@[to_additive "An `AddSubmonoid` of an `AddMonoid` inherits a zero."]
+instance one : One S' :=
   ⟨⟨1, OneMemClass.one_mem S'⟩⟩
-#align one_mem_class.has_one OneMemClass.hasOne
+#align one_mem_class.has_one OneMemClass.one
+#align zero_mem_class.has_zero ZeroMemClass.zero
 
 @[simp, norm_cast, to_additive]
 theorem coe_one : ((1 : S') : M₁) = 1 :=
   rfl
 #align one_mem_class.coe_one OneMemClass.coe_one
+#align zero_mem_class.coe_zero ZeroMemClass.coe_zero
 
 variable {S'}
 
 @[simp, norm_cast, to_additive]
 theorem coe_eq_one {x : S'} : (↑x : M₁) = 1 ↔ x = 1 :=
   (Subtype.ext_iff.symm : (x : M₁) = (1 : S') ↔ x = 1)
 #align one_mem_class.coe_eq_one OneMemClass.coe_eq_one
+#align zero_mem_class.coe_eq_zero ZeroMemClass.coe_eq_zero
 
 variable (S')
 
 @[to_additive]
 theorem one_def : (1 : S') = ⟨1, OneMemClass.one_mem S'⟩ :=
   rfl
 #align one_mem_class.one_def OneMemClass.one_def
+#align zero_mem_class.zero_def ZeroMemClass.zero_def
 
 end OneMemClass
 
-namespace SubmonoidClass
-
 variable {A : Type _} [SetLike A M] [hA : SubmonoidClass A M] (S' : A)
 
-/-- An `add_submonoid` of an `add_monoid` inherits a scalar multiplication. -/
-instance AddSubmonoidClass.hasNsmul {M} [AddMonoid M] {A : Type _} [SetLike A M]
-    [AddSubmonoidClass A M] (S : A) : HasSmul ℕ S :=
-  ⟨fun n a => ⟨n • a.1, nsmul_mem a.2 n⟩⟩
-#align add_submonoid_class.has_nsmul AddSubmonoidClass.hasNsmul
+/-- An `AddSubmonoid` of an `AddMonoid` inherits a scalar multiplication. -/
+instance AddSubmonoidClass.SMul {M} [AddMonoid M] {A : Type _} [SetLike A M]
```

Comment:
You've changed the autogenerated instance name from `.hasNsmul` to `.SMul`.  In mathlib3 `nsmul` had a particular meaning, referring to the automatic "scalar multiplication" by natural numbers one has in a monoid.  I think therefore that we should keep the extra information indicated by the "n".  Not sure if we've fixed a naming/capitalization convention for this yet -- maybe `.nSMul`?

Same on `SubmonoidClass.Pow`.  (But we were actually inconsistent in mathlib3, referring to `nsmul` but to `pow` rather than `npow`.)

Reply:
Thanks for all the remarks! Followed them as best as I could :)